src/enzo/SetEvolveRefineRegion.C

```diff
@@ -23,68 +27,281 @@ int SetEvolveRefineRegion (FLOAT time)
 
   /* Return if not used */
 
-  if (EvolveRefineRegionTime[0] <= FLOAT_UNDEFINED)
+  if (EvolveRefineRegionTime[0] <= FLOAT_UNDEFINED
+      && EvolveMustRefineRegionTime[0] <= FLOAT_UNDEFINED
+      && EvolveCoolingRefineRegionTime[0] <= FLOAT_UNDEFINED)
     return SUCCESS;
 
+  if(debug1){
+    printf("SetEvolveRefineRegion: setting region.\n");
+    printf("SetEvolveRefineRegion: EvolveRefineRegionTime[0]:        %"FSYM"\n",EvolveRefineRegionTime[0]);
+    printf("SetEvolveRefineRegion: EvolveMustRefineRegionTime[0]:    %"FSYM"\n",EvolveMustRefineRegionTime[0]);
+    printf("SetEvolveRefineRegion: EvolveCoolingRefineRegionTime[0]: %"FSYM"\n",EvolveCoolingRefineRegionTime[0]);
+  }
+  
   /* If TimeType is redshift, calculate redshift */
 
   if (ComovingCoordinates) {
     CosmologyComputeExpansionFactor(time, &a, &dadt);
     redshift = (1 + InitialRedshift)/a - 1;
   }
 
-  /* Find closest time step with <time */
-  /* Set time=redshift if that's what we're doing. */
-  if (RefineRegionTimeType == 1) {
-    time = redshift;
-    for(timestep=0; timestep<EvolveRefineRegionNtimes; timestep++){
-      if( time > EvolveRefineRegionTime[timestep] ){
-        break;
+  /* does time evolution for standard refinement regions */
+  if((RefineRegionTimeType == 1) || (RefineRegionTimeType == 0) ){
+  
+    /* Find closest time step with <time */
+    /* Set time=redshift if that's what we're doing. */
+    if (RefineRegionTimeType == 1) {  // redshift
+
+      time = redshift;
+
+      /* Check to see if the current redshift is within the bounds of the given time file.  This assumes
+	 that EvolveRefineRegionTime[0] is the highest redshift and the last output is the lowest (for the 
+	 given track file).  */
+      if(time > EvolveRefineRegionTime[0] || time < EvolveRefineRegionTime[EvolveRefineRegionNtimes-1]){
+	fprintf(stderr,"SetEvolveRefineRegion ERROR: current simulation redshift is outside of range of track file redshifts!");
+	my_exit(EXIT_FAILURE);
+      }
+
+      for(timestep=0; timestep<EvolveRefineRegionNtimes; timestep++){
+	if( time > EvolveRefineRegionTime[timestep] ){
+	  break;
+	}
+      }
+    }else{  // code time
+
+      /* Check to see if the current time is within the bounds of the given time file.  This assumes
+	 that EvolveRefineRegionTime[0] is the earliest time and the last output is the latest time (for
+	 the given track file). */
+      if(time < EvolveRefineRegionTime[0] || time > EvolveRefineRegionTime[EvolveRefineRegionNtimes-1]){
+	fprintf(stderr,"SetEvolveRefineRegion ERROR: current simulation time is outside of range of track file redshifts!");
```

Comment:
This print statement should say "times" instead of "redshifts".
```suggestion
	fprintf(stderr,"SetEvolveRefineRegion ERROR: current simulation time is outside of range of track file times!");
```

Reply:
Fixed!